pkg/metrics/handler.go

```diff
@@ -0,0 +1,51 @@
+package metrics
+
+import (
+	"net/http"
+
+	"github.com/rancher/rancher/pkg/auth/util"
+	v2 "k8s.io/api/authorization/v1"
+	v3 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/kubernetes/typed/authorization/v1"
+)
+
+type metricsHandler struct {
```

Comment:
How about a comment that says...

```go
metricsHandler is an HTTP middleware that verifies that the user making the request can get  ranchermetrics resources before calling the provided handler.

This uses the Impersonate-User and Impersonate-Groups HTTP headers to check access.
```

Also, there's nothing specific to prometheus here, so maybe we drop the references to Prometheus and `prom` in the parameters and struct field?